### PR TITLE
[feat]: [PIE-9226]: Handled exception related to template resolution for Async Validation to Unblock User

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Issue: Whenever there was an error thrown in template resolve from Ng-manager side we were not handling it and validation status was not being updated.
Change: Have handled error and updated validation status to false.
Testing: Local.

Screenshot 2023-03-28 at 4 53 33 PM

As there is no issues reported in the list: we have this:
Screenshot 2023-03-28 at 5 09 56 PM

Note: This PR is only for unblocking the user, proper handling will be done in the next sprint.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)

### CODE REVIEW

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 80 at r1](https://reviewable.io/reviews/harness/harness-core/45751#-NRc01d43maiRUCR6HqJ:-NRc01d43maiRUCR6HqK:b-97q9bn) ([raw file](https://github.com/harness/harness-core/blob/81a3ba4375972f8fb0528f3aab227f169abbb51a/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L80)):*
> ```Java
>               .templateInputsResponse(ValidateTemplateInputsResponseDTO.builder().validYaml(false).build())
>               .build();
>       validationService.updateEvent(validationEvent.getUuid(), ValidationStatus.FAILURE, templateValidationResult);
> ```

what if this method call fails? Is it fine to simply let it throw the exception back?



_Code quote:_
```Java
updateEvent
```

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 46 at r2](https://reviewable.io/reviews/harness/harness-core/45751#-NRc4GgQ1qPYxBC4H6iv:-NRc4GgR8_lvTXP-_Ywi:b-ad5ee0) ([raw file](https://github.com/harness/harness-core/blob/6aa0cc279c27737cc7613e9dbe5d48c135078f6d/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L46)):*
> ```Java
> 
>   @Override
>   public void run() {
> ```

Can you try throwing exception in run method and see pipeline get operations works?
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 78 at r2](https://reviewable.io/reviews/harness/harness-core/45751#-NRc4dUnDqkfctxGhdq8:-NRc4dUnDqkfctxGhdq9:b-2fq6ky) ([raw file](https://github.com/harness/harness-core/blob/6aa0cc279c27737cc7613e9dbe5d48c135078f6d/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L78)):*
> ```Java
>       templateValidationResult =
>           ValidationResult.builder()
>               .templateInputsResponse(ValidateTemplateInputsResponseDTO.builder().validYaml(false).build())
> ```

we should not use templateInputsResponse body for template as its for template-service one, we should create new field.

{exceptionMessage for templateInputs failure only for now}

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 78 at r2](https://reviewable.io/reviews/harness/harness-core/45751#-NRc4dUnDqkfctxGhdq8:-NRc5sXV5v2FTI2e-MXM:brsx6fn) ([raw file](https://github.com/harness/harness-core/blob/6aa0cc279c27737cc7613e9dbe5d48c135078f6d/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L78)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

we should not use templateInputsResponse body for template as its for template-service one, we should create new field.

{exceptionMessage for templateInputs failure only for now}
</blockquote></details>

You will need to update PipelineValidationResponseBody (for new API) and PipelineValidationResponseDTO for existing API to propagate this template error to API.

UI also needs to understand that

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 9 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 80 at r1](https://reviewable.io/reviews/harness/harness-core/45751#-NRc01d43maiRUCR6HqJ:-NRkok960Gom5NWkPgnj:bwo4vn6) ([raw file](https://github.com/harness/harness-core/blob/81a3ba4375972f8fb0528f3aab227f169abbb51a/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L80)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

what if this method call fails? Is it fine to simply let it throw the exception back?
</blockquote></details>

Yes, this just a method with inly DB operation and this while process is an async one if we are not able write to DB, it will have same effect as this method never been hit on UI.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 46 at r2](https://reviewable.io/reviews/harness/harness-core/45751#-NRc4GgQ1qPYxBC4H6iv:-NRkoXN-EsU5JqjNesV8:b2gk9kf) ([raw file](https://github.com/harness/harness-core/blob/6aa0cc279c27737cc7613e9dbe5d48c135078f6d/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L46)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Can you try throwing exception in run method and see pipeline get operations works?
</blockquote></details>

I had set threadpool size to 1 and we were able t hit the run method again if the earlier threw exception.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 78 at r2](https://reviewable.io/reviews/harness/harness-core/45751#-NRc4dUnDqkfctxGhdq8:-NRkojOT-6QtLLsblcSk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6aa0cc279c27737cc7613e9dbe5d48c135078f6d/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L78)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

You will need to update PipelineValidationResponseBody (for new API) and PipelineValidationResponseDTO for existing API to propagate this template error to API.

UI also needs to understand that
</blockquote></details>

Done.

**Review by archit-harness — DISMISSED** [review]
> Reviewed 2 of 2 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 11 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 555 at r4](https://reviewable.io/reviews/harness/harness-core/45751#-NRle6gi3cw-2gHHGM4U:-NRm_y_xE5pOLHO2IydK:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/924e19dcbef77d1e43dbc7c5e34d7a100f62bd83/pipeline-service/contracts/openapi/v1/openapi.yaml#L555)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

we should not change the order
</blockquote></details>

done

**Review by shalini-agr — APPROVED** [review]
> Reviewed 2 of 8 files at r3, 4 of 4 files at r4, 2 of 2 files at r5, 1 of 2 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 2 file(s) to review. 1. unresolved discussions.

**Comment by senjucanon2** [comment]
trigger ut0, ut4

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java` line 80 at r1](https://reviewable.io/reviews/harness/harness-core/45751#-NRc01d43maiRUCR6HqJ:-NRnixVp3xKriesK2jPg:bcanltl) ([raw file](https://github.com/harness/harness-core/blob/81a3ba4375972f8fb0528f3aab227f169abbb51a/pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java#L80)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

Yes, this just a method with inly DB operation and this while process is an async one if we are not able write to DB, it will have same effect as this method never been hit on UI.
</blockquote></details>

What if it throws error? What happens next? Do we handle it elegantly in downstream layer at the caller?

**Review by mohitg0795 — DISMISSED** [review]
> Reviewed 2 of 8 files at r3, 4 of 4 files at r4, 2 of 2 files at r5, 1 of 2 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: :shipit: complete! ✓

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1955,7 +1957,18 @@ components:
...
           type: string
-      description: 'Response body for moving an input set configuration.'
+      description: Response body for moving an input set configuration.
+    TemplateValidationResponseBody:
+      title: TemplateValidationResponseBody
+      x-stoplight:
+        id: fulzeg8b945v8
+      type: object
+      properties:
+        validYaml:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1967]: valid_yaml

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -1955,7 +1957,18 @@ components:
...
+          type: boolean
+        exceptionMessage:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1969]: exception_message

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 1 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1967 at r7](https://reviewable.io/reviews/harness/harness-core/45751#-NRkoXqS8Bio3fM6HhLw-r7-1967:-NRqGK5U8bdM3qrs7Y1f:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a416d2874ab08c6dac17c761867199418369b66f/pipeline-service/contracts/openapi/v1/openapi.yaml#L1967)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

valid_yaml
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1969 at r7](https://reviewable.io/reviews/harness/harness-core/45751#-NRkoXqS8Bio3fM6HhLw-r7-1969:-NRqGKfGDMh1uW892_su:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a416d2874ab08c6dac17c761867199418369b66f/pipeline-service/contracts/openapi/v1/openapi.yaml#L1969)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

exception_message
</blockquote></details>

Done.

**Review by mankrit-singh — COMMENTED** [review]
> Reviewed 2 of 8 files at r3, 4 of 4 files at r4, 2 of 2 files at r5, 1 of 2 files at r6, 1 of 2 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 1 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by mankrit-singh — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger ut3

**Review by shalini-agr — DISMISSED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: :shipit: complete! ✓

**Comment by senjucanon2** [comment]
trigger ut2, ut3, ut6

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by shalini-agr — APPROVED** [review]
> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: :shipit: complete! ✓

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r7, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45751)* status: :shipit: complete! ✓

**Pull request merged**